presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -601,10 +585,10 @@ protected Void visitSubqueryExpression(SubqueryExpression node, Void context)
             return null;
         }
 
-        private void verifyField(Expression node, QualifiedName qualifiedName)
+        private void isColumnReferenceUseCorrect(Expression node)
         {
-            ResolvedField resolvedField = orderByScope.get().resolveField(node, qualifiedName);
-            if (resolvedField.getScope().equals(orderByScope.get())) {
+            FieldId fieldId = requireNonNull(columnReferences.get(node), () -> "No FieldId for " + node);
+            if (Objects.equals(orderByScope.get().getRelationType().getRelationId(), fieldId.getRelationId())) {
```

Comment:
I was not sure about semantics of such a method (what shall it do with wrong field index? name suggests `false`, but I'd prefer an exception), so I simply extracted this condition to a meaningfully named method in here.